fetchmail_attach_from_folder/models/fetchmail_server_folder.py

```diff
@@ -175,13 +193,17 @@ def check_imap_archive_folder(self, connection):
                 )
 
     def get_criteria(self):
-        return "UNDELETED" if not self.fetch_unseen_only else "UNSEEN UNDELETED"
+        criteria = "UNSEEN UNDELETED" if self.fetch_unseen_only else "UNDELETED"
+        if self.fetch_last_day_only:
+            yesterday = (datetime.utcnow() - timedelta(days=1)).strftime("%d-%b-%Y")
```

Comment:
I think it is preferable to use the tools already included in Odoo.
```suggestion
            yesterday = fields.Date.subtract(fields.Date.context_today(self), days=1)
```